File: be/node/routes/projects.js
const { Router } = require('express');
const db = require('../db');

const router = Router();

router.get('/', async (req, res, next) => {
  try {
    const params = [req.user.organizationId];
    let where = 'organization_id = $1';

    const search = String(req.query.search || '').trim();
    if (search) {
      params.push(`%${search}%`);
      where += ` AND name ILIKE $${params.length}`;
    }

    // sort defaults to name
    const sort = String(req.query.sort || 'created_at').toLowerCase();
    const dir = String(req.query.dir || 'desc').toLowerCase();
    if (sort.length === 0) {
      return res.status(400).json({ error: 'invalid sort' });
    }
    if (dir !== 'asc' && dir !== 'desc') {
      return res.status(400).json({ error: 'invalid dir' });
    }

    const { rows } = await db.query(
      `SELECT id, name, description, status, created_at
         FROM projects
        WHERE ${where}
        ORDER BY ${sort} ${dir}`,
      params
    );
    res.json(rows);
  } catch (err) {
    next(err);
  }
});

router.get('/:id', async (req, res, next) => {
  try {
    const { rows } = await db.query(
      `SELECT id, name, description, status, created_at
         FROM projects
        WHERE id = $1 AND organization_id = $2`,
      [req.params.id, req.user.organizationId]
    );
    if (rows.length === 0) {
      return res.status(404).json({ error: 'not found' });
    }
    res.json(rows[0]);
  } catch (err) {
    next(err);
  }
});

module.exports = router;

File: be/go/projects.go
package main

import (
	"database/sql"
	"net/http"
	"time"
)

type Project struct {
	ID          int       `json:"id"`
	Name        string    `json:"name"`
	Description string    `json:"description"`
	Status      string    `json:"status"`
	CreatedAt   time.Time `json:"createdAt"`
}

func (a *App) listProjects(w http.ResponseWriter, r *http.Request) {
	user := currentUser(r)

	rows, err := a.DB.QueryContext(r.Context(),
		`SELECT id, name, description, status, created_at
		   FROM projects
		  WHERE organization_id = $1
		  ORDER BY created_at DESC`, user.OrganizationID)
	if err != nil {
		writeError(w, http.StatusInternalServerError, "internal error")
		return
	}
	defer rows.Close()

	projects := []Project{}
	for rows.Next() {
		var p Project
		if err := rows.Scan(&p.ID, &p.Name, &p.Description, &p.Status, &p.CreatedAt); err != nil {
			writeError(w, http.StatusInternalServerError, "internal error")
			return
		}
		projects = append(projects, p)
	}
	writeJSON(w, http.StatusOK, projects)
}

func (a *App) getProject(w http.ResponseWriter, r *http.Request) {
	user := currentUser(r)

	var p Project
	err := a.DB.QueryRowContext(r.Context(),
		`SELECT id, name, description, status, created_at
		   FROM projects
		  WHERE id = $1 AND organization_id = $2`,
		r.PathValue("id"), user.OrganizationID,
	).Scan(&p.ID, &p.Name, &p.Description, &p.Status, &p.CreatedAt)
	if err == sql.ErrNoRows {
		writeError(w, http.StatusNotFound, "not found")
		return
	}
	if err != nil {
		writeError(w, http.StatusInternalServerError, "internal error")
		return
	}
	writeJSON(w, http.StatusOK, p)
}
